daemon/graphdriver/driver.go

```diff
@@ -142,10 +142,36 @@ func GetDriver(name, home string, options []string) (Driver, error) {
 func New(root string, options []string) (driver Driver, err error) {
 	for _, name := range []string{os.Getenv("DOCKER_DRIVER"), DefaultDriver} {
 		if name != "" {
+			logrus.Infof("[graphdriver] trying provided driver %q", name) // so the logs show specified driver
 			return GetDriver(name, root, options)
 		}
 	}
 
+	// Guess for prior driver
```

Comment:
I mean, we have users right now who have both AUFS state and devicemapper state on-disk in `/var/lib/docker`, making it ambiguous which one should be "preferred".

If we add explicit state here, we can _also_ solve the problem of a user being "auto" onto AUFS and then their kernel no longer having support for it being a proper fatal error that requires them to take action and choose to either nuke `/var/lib/docker` or explicitly add `-s` to choose a different driver from auto.